Cattac/Cattac/GameEngine.swift

```diff
@@ -150,13 +161,22 @@ class GameEngine {
         }
     }
     
+    /// Releases all resources associated with this game, called
+    /// when game has ended.
     func end() {
         for ref in movementWatchers.values {
             ref.removeAllObservers()
         }
     }
     
+    /// Trigger state advancement in game engine.
     private func nextState() {
```

Comment:
Should we still keep this method? Seems kind of pointless here since all it does is changing `statesToAdvance`